notifications.tf

```diff
@@ -0,0 +1,48 @@
+locals {
+    # Whether to enable notifications or not
+    enable_notifications = var.enabled && length(var.notifications_backup_vault_events) > 0 ? true : false
+}
+
+resource "aws_sns_topic" "backup_events" {
+  count  = local.enable_notifications ? 1 : 0
+  name = var.notifications_sns_topic_name
+}
+
+data "aws_iam_policy_document" "backup_events" {
+  statement {
+    actions = [
+      "SNS:Publish",
+    ]
+    effect = "Allow"
+    principals {
+      type        = "Service"
+      identifiers = ["backup.amazonaws.com"]
+    }
+    resources = [
+      local.enable_notifications ? aws_sns_topic.backup_events[0].arn : "",
+    ]
+    sid = "BackupPublishEvents"
+  }
+}
+
+resource "aws_sns_topic_policy" "backup_events" {
+  count  = local.enable_notifications ? 1 : 0
+  arn    = aws_sns_topic.backup_events[0].arn
+  policy = data.aws_iam_policy_document.backup_events.json
+}
+
+resource "aws_backup_vault_notifications" "backup_events" {
+  count               = local.enable_notifications ? 1 : 0
+  backup_vault_name   = var.vault_name != null ? var.vault_name : "Default"
+  sns_topic_arn       = aws_sns_topic.backup_events[0].arn
```

Comment:
Once the SNS topic was created outside this module, pass the ARN here, using the SNS resource or module output as the input of this module